### PR TITLE
[Lib] Add `connectible.remote` class

### PR DESCRIPTION
**Pull request by vamahaja**
Add `cli/connectible/remote.py` library to support ssh client operations which include -

1. `_connect` - Connect to host.
2. `run` - Execute command on host.
3. `run_async` - Execute command in background on host.
4. `class RemoteConnectionError` - An exception for failed operations.

### CODE REVIEW

File: cli/connectible/remote.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        if w.expired:
+            raise UnexpectedStateError(
+                f"Failed to complete command within {timeout} sec"
+            )
+
+        # Read command error
+        while True:
+            try:
+                # Get 1kb of error
+                err = channel.recv_stderr(1024)
```
Comment (pranavprakash20) [inline cli/connectible/remote.py:137]: One doubt here, do we need to add`.decode('ascii')`  for the data being received? or it is already in readable format?

File: cli/connectible/remote.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+from cli.exceptions import RemoteConnectionError, UnexpectedStateError
+from cli.utilities.waiter import WaitUntil
+from utility.log import Log
+
+LOG = Log(__name__)
+
+
+class Remote:
+    """Insterface for SSH operations"""
```
[the remark this answers is not in the document]
Comment (vamahaja) [reply cli/connectible/remote.py:13]: Good catch

File: cli/connectible/remote.py (hunk 1 above)
Comment (vamahaja) [reply cli/connectible/remote.py:137]: `recv_stderr` returns data in bytes. `err` is validated and converted to string at line #142

**Review by pranavprakash20 — APPROVED** [review]
Lgtm

**Review by MohitBis — APPROVED** [review]
LGTM

**Review by tintumathew10 — APPROVED** [review]
LGTM

**Pull request merged**